### PR TITLE
Truffle fixes

### PR DESCRIPTION
**Pull request by JcBernack**

### CODE REVIEW

File: libevm/contracts/ForgerStakes.sol [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: GPL-3.0
+// SPDX-License-Identifier: MIT
```
Comment (i-Alex) [inline libevm/contracts/ForgerStakes.sol:1]: a comment for the `struct Stake` below.
It is a bit inconsistent with the actual one: vrfPublicKey is 33 bytes

File: sdk/src/main/scala/com/horizen/account/receipt/EthereumLog.scala [hunk 2]
```diff
@@ -1,60 +0,0 @@
...
-
-import com.horizen.evm.interop.EvmLog
-import com.horizen.evm.utils.{Address, Hash}
-import com.horizen.utils.{ByteArrayWrapper, BytesUtils}
-import scorex.core.serialization.ScorexSerializer
-import scorex.util.serialization.{Reader, Writer}
-
-import java.util
-
-case class EthereumLog(
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/account/receipt/EthereumLog.scala:11]: @alsala @JcBernack let's discuss the usage of this class.

File: libevm/contracts/ForgerStakes.sol (hunk 1 above)
Comment (JcBernack) [reply libevm/contracts/ForgerStakes.sol:1]: True, but unfortunately there is no type `bytes33` in Solidity. The ABI serializer on our end also handles it like that, see [com.horizen.account.state.ForgerPublicKeys#vrfPublicKeyToAbi](https://github.com/HorizenOfficial/Sidechains-SDK/blob/d1180fd0160dd4adbd65de570fb2a3e293d8b2ea/sdk/src/main/scala/com/horizen/account/state/ForgerStakeMsgProcessor.scala#L460).

Comment (JcBernack) [reply libevm/contracts/ForgerStakes.sol:1]: Oops, I mixed up the files. This is just my draft for the Forger Stakes contract before we decided to implement them in Java. This is not used anywhere and could be removed.

Comment (JcBernack) [reply libevm/contracts/ForgerStakes.sol:1]: I was talking about the Forger Stakes Contract interface in Solidity where the vrf public key is separated to a `bytes32` property and then one more `bytes1`.

File: sdk/src/main/scala/com/horizen/account/state/ForgerStakeMsgProcessor.scala [hunk 3]
```diff
@@ -108,18 +116,8 @@ case class ForgerStakeMsgProcessor(params: NetworkParams) extends FakeSmartContr
...
-      val modPreviousNode = LinkedListNode(
-        previousNode.dataKey,
-        previousNode.previousNodeKey,
-        newTip
-      )
-
-      // store the modified previous node
-      view.updateAccountStorageBytes(contractAddress, oldTip,
-        LinkedListNodeSerializer.toBytes(modPreviousNode))
+    modifyNode(view, oldTip) { previousNode =>
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/account/state/ForgerStakeMsgProcessor.scala:119]: nice refactoring!

**Review by i-Alex — APPROVED** [review]
Seems good!
Note: just a comment about the outdated fake smart contract solidity code. Let's remove them at all.

File: libevm/contracts/ForgerStakes.sol (hunk 1 above)
Comment (JcBernack) [reply libevm/contracts/ForgerStakes.sol]: done

**Review by i-Alex — APPROVED** [review]

**Review by akrckn — APPROVED** [review]

**Pull request merged**